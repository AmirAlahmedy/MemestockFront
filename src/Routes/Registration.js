import React from 'react';
import './Registration.css';




const Registration = props => {

    
        return(
           
        <div>

            <form className='regForm' onSubmit={ props.regHand }>
                <h1 className='logo'>Memestock</h1>
                <div className='inputs'>

                <input type='text' 
                placeholder='E-mail'/>

                <input type='text' 
                placeholder='User name'/>

                <input type="password" 
                placeholder="Password" 
                value={ props.password } 
                onChange={ props.svPswrd }/>

                </div>
                <button type="submit" className='registerButton'> Register </button>
            </form>
               
        </div>
               );

    
}

export default Registration;