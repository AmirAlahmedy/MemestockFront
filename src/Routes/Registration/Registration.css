/* $redditOrange: #ff4500;
$redditBlack: rgb(26, 26, 27); */
.regForm {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  text-align: center;
  min-height: 70vh;
  width: 50%;
  background-color: #1a1a1b;
  padding-top: 50px;
  border: 1px solid #343536;
  border-radius: 6px;
}

.logo {
  font-style: italic;
  color: #f2f2f2;
  font-size: 32px;
  cursor: pointer;
}

.inputs {
  padding-top: 20px;
}

.inputs input {
  width: 60%;
  -webkit-transition: width 0.4s ease-in-out;
  transition: width 0.4s ease-in-out;
  border-radius: 20px;
  outline: none;
  border: 1px solid #ccc;
  background-color: white;
  font: inherit;
  padding: 6px 10px;
  display: block;
  width: 100%;
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
}

.usernameInput, .emailInput {
  margin-bottom: 30px;
}

input:focus {
  /* border: 3px solid #555; */
  width: 70%;
  outline: none;
  background-color: #ccc;
}

.registerButton {
  margin-top: 30px;
  background-color: #ff4500;
  border-radius: 20px;
  font-family: inherit;
  font-size: 18px;
  color: #f2f2f2;
  padding: 10px;
  text-shadow: 0 0.04em 0.04em rgba(0, 0, 0, 0.35);
  border-style: initial;
  -webkit-transition: background .3s,color .3s;
  transition: background .3s,color .3s;
  -webkit-box-shadow: 0px 0px 0px 03px  #ff4400c7;
          box-shadow: 0px 0px 0px 03px  #ff4400c7;
  -webkit-transition: -webkit-box-shadow 1s linear;
  transition: -webkit-box-shadow 1s linear;
  transition: box-shadow 1s linear;
  transition: box-shadow 1s linear, -webkit-box-shadow 1s linear;
  margin: 0.5em;
}

.registerButton:hover {
  background-color: #f2f2f2;
  /* border: 2px solid #ff4400c7;   */
  color: #ff4400c7;
  -webkit-box-shadow: 0px 0px 0px 6px #ff4400c7;
          box-shadow: 0px 0px 0px 6px #ff4400c7;
}

.formWrapper {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  padding-top: 5%;
}
/*# sourceMappingURL=Registration.css.map */