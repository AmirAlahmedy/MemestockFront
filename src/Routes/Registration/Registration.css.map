{
    "version": 3,
    "mappings": "AAAA;iCACiC;AACjC,AAAA,QAAQ,CAAC;EACL,OAAO,EAAE,IAAI;EACb,cAAc,EAAE,MAAM;EACtB,WAAW,EAAE,MAAM;EACnB,UAAU,EAAE,MAAM;EAClB,UAAU,EAAE,IAAI;EAChB,KAAK,EAAE,GAAG;EACV,gBAAgB,EAAE,OAAe;EACjC,WAAW,EAAE,IAAI;EACjB,MAAM,EAAE,GAAG,CAAC,KAAK,CAAC,OAAe;EACjC,aAAa,EAAE,GAAG;CACrB;;AACD,AAAA,KAAK,CAAC;EACF,UAAU,EAAE,MAAM;EAClB,KAAK,EAAE,OAAO;EACd,SAAS,EAAE,IAAI;EACf,MAAM,EAAE,OAAO;CAClB;;AACD,AAAA,OAAO,CAAC;EACJ,WAAW,EAAE,IAAI;CACpB;;AACD,AAAA,OAAO,CAAC,KAAK,CAAA;EACT,KAAK,EAAE,GAAG;EACV,UAAU,EAAE,sBAAsB;EAClC,aAAa,EAAE,IAAI;EAGnB,OAAO,EAAE,IAAI;EACb,MAAM,EAAE,cAAc;EACtB,gBAAgB,EAAE,KAAK;EACvB,IAAI,EAAE,OAAO;EACb,OAAO,EAAE,QAAQ;EACjB,OAAO,EAAE,KAAK;EACd,KAAK,EAAE,IAAI;EACX,UAAU,EAAE,UAAU;CACzB;;AACD,AAAA,cAAc,EAAE,WAAW,CAAA;EACvB,aAAa,EAAE,IAAI;CACtB;;AACD,AAAA,KAAK,AAAA,MAAM,CAAC;EAER,6BAA6B;EAC7B,KAAK,EAAE,GAAG;EAEV,OAAO,EAAE,IAAI;EACb,gBAAgB,EAAE,IAAI;CACvB;;AACH,AAAA,eAAe,CAAC;EACZ,UAAU,EAAE,IAAI;EAChB,gBAAgB,EAAE,OAAO;EACzB,aAAa,EAAE,IAAI;EACnB,WAAW,EAAE,OAAO;EACpB,SAAS,EAAE,IAAI;EACf,KAAK,EAAE,OAAO;EACd,OAAO,EAAE,IAAI;EACb,WAAW,EAAE,CAAC,CAAC,MAAM,CAAC,MAAM,CAAC,mBAAgB;EAC7C,YAAY,EAAE,OAAO;EACrB,UAAU,EAAE,wBAAwB;EAEpC,UAAU,EAAE,2BAA2B;EACvC,UAAU,EAAE,oBAAoB;EAChC,MAAM,EAAE,KAAK;CAEhB;;AACD,AAAA,eAAe,AAAA,MAAM,CAAA;EACjB,gBAAgB,EAAE,OAAO;EACzB,oCAAoC;EACpC,KAAK,EAAE,SAAS;EAChB,UAAU,EAAE,yBAAyB;CACxC;;AACC,AAAA,YAAY,CAAA;EACT,OAAO,EAAE,IAAI;EACb,cAAc,EAAE,MAAM;EACtB,WAAW,EAAE,MAAM;EACnB,WAAW,EAAE,EAAE;CAGlB",
    "sources": [
        "Registration.scss"
    ],
    "names": [],
    "file": "Registration.css"
}