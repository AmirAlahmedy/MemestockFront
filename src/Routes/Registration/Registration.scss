/* $redditOrange: #ff4500;
$redditBlack: rgb(26, 26, 27); */
.regForm {
    display: flex;
    flex-direction: column;
    align-items: center;
    text-align: center;
    min-height: 70vh;
    width: 50%;
    background-color: rgb(26, 26, 27);
    padding-top: 50px;
    border: 1px solid rgb(52, 53, 54);
    border-radius: 6px;
}
.logo {
    font-style: italic;
    color: #f2f2f2;
    font-size: 32px;
    cursor: pointer;
}
.inputs {
    padding-top: 20px;
}
.inputs input{
    width: 60%;
    transition: width 0.4s ease-in-out;
    border-radius: 20px;
    

    outline: none;
    border: 1px solid #ccc;
    background-color: white;
    font: inherit;
    padding: 6px 10px;
    display: block;
    width: 100%;
    box-sizing: border-box;
}
.usernameInput, .emailInput{
    margin-bottom: 30px; 
}
input:focus {
    
    /* border: 3px solid #555; */
    width: 70%;

    outline: none;
    background-color: #ccc;
  }
.registerButton {
    margin-top: 30px;
    background-color: #ff4500;
    border-radius: 20px;
    font-family: inherit;
    font-size: 18px;
    color: #f2f2f2;
    padding: 10px;
    text-shadow: 0 0.04em 0.04em rgba(0,0,0,0.35);
    border-style: initial;
    transition: background .3s,color .3s;

    box-shadow: 0px 0px 0px 03px  #ff4400c7;
    transition: box-shadow 1s linear;
    margin: 0.5em;
    
}
.registerButton:hover{
    background-color: #f2f2f2;
    /* border: 2px solid #ff4400c7;   */
    color: #ff4400c7; 
    box-shadow: 0px 0px 0px 6px #ff4400c7;
}
  .formWrapper{
     display: flex;
     flex-direction: column;
     align-items: center;
     padding-top: 5%;
    

 } 

