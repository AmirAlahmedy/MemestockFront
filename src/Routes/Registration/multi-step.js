// import MultiStep from 'react-multistep';
// import React, { Component } from 'react';

// class multistep extends Component {
//      steps = [
//         {name: 'StepOne', component: <StepOne/>},
//         {name: 'StepTwo', component: <StepTwo/>},
//         {name: 'StepThree', component: <StepThree/>},
//         {name: 'StepFour', component: <StepFour/>}
//       ];
//       render(){
//           return(
//           <Multistep showNavigation={true} steps={steps}/>
//           )
//       }
// }
// export default multistep;