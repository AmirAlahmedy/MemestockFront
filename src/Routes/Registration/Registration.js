import React, { Component } from 'react';
import './Registration.css';
import Aux from '../../Components/HOC/Auxiliary';
import classes from './Registration.module.css';
import { NavLink, Route, withRouter } from 'react-router-dom';
import Login from '../Login/Login';



class Registration extends Component {

    componentDidMount = () =>{
        this.props.history.replace('/Registration/');
    }
    
    render(){

        
        let error = <p className={classes.Invalid}>Your password is wrong!</p>
        return(

        <Aux>

            <form className='regForm' onSubmit={ this.props.regHand }>
                <h1 className='logo'>Memestock</h1>
                 <div className='inputs'>

                <input type='email' 
                placeholder='E-mail'
                name='email'/>

                <input type='text' 
                placeholder='User name'
                name='username'/>

                <input type='password'
                placeholder="Password" 
                // value={ this.props.password } 
                onChange={ this.props.svPswrd }
                name='password'/>

                {!this.props.logged && !this.props.psrdVld ? error:null}
                </div> 

                <button type="submit" className='registerButton'> Register </button>
                <NavLink to='/Login/'>Already Registered?</NavLink>
            </form>
            <Route path='/Login/'  render={()=><Login logHand={this.props.regHand} logged={this.props.logged} psrdVld={this.props.psrdVld} password={this.props.password} svPswrd={this.props.svPswrd}/>}/>

        </Aux>
               );
               
               
            }
               
        }
            
export default withRouter(Registration);