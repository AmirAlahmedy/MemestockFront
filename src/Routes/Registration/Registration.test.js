import React from 'react';
import axios from 'axios';
import { shallow } from 'enzyme';
import Registration from '../Registration/Registration';

jest.mock('axios');

describe('Registration component', () => {
    describe('when rendered', () => {
        interface('should fetch a post request', () => {
            const postSpy = jest.spyOn(axios, 'post');
            const registratinonInstance = shallow(
                <Registration/>
            )
        });
        expect(postSpy).toBeCalled();
    })
})