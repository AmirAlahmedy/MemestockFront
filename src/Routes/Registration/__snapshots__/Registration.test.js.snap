// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Registration /> it matches the snapshot 1`] = `
<div
  className="formWrapper"
>
  <form
    className="regForm"
    onSubmit={[Function]}
  >
    <h1
      className="logo"
    >
      SIGN UP
    </h1>
    <p
      className="Invalid"
    >
      
    </p>
    <div
      className="inputs"
    >
      <input
        className="emailInput in"
        name="email"
        placeholder="E-mail"
        type="email"
      />
      <input
        className="usernameInput in"
        name="username"
        placeholder="User name"
        type="text"
      />
      <input
        className="passInput in"
        name="password"
        placeholder="Password"
        type="password"
      />
    </div>
    <button
      className="registerButton"
      type="submit"
    >
      SIGN UP
    </button>
  </form>
</div>
`;
